src/ServiceInsight.Tests/MessageListViewModelTests.cs

```diff
@@ -28,39 +29,40 @@ public class MessageListViewModelTests
         public void TestInitialize()
         {
             eventAggregator = Substitute.For<IEventAggregator>();
+            workNotifier = Substitute.For<IWorkNotifier>();
             serviceControl = Substitute.For<IServiceControl>();
             searchBar = Substitute.For<SearchBarViewModel>();
             clipboard = Substitute.For<IClipboard>();
-            messageListFunc = () => new MessageListViewModel(eventAggregator,
-                                                   serviceControl,
-                                                   searchBar,
-                                                   Substitute.For<GeneralHeaderViewModel>(),
-                                                   Substitute.For<MessageSelectionContext>(),
-                                                   clipboard);
+            messageListFunc = () => new MessageListViewModel(
+                eventAggregator,
+                workNotifier,
+                serviceControl,
+                searchBar,
+                Substitute.For<GeneralHeaderViewModel>(),
+                Substitute.For<MessageSelectionContext>(),
+                clipboard);
         }
 
         [Test]
         public void Should_load_the_messages_from_the_endpoint()
         {
             var endpoint = new Endpoint { Host = "localhost", Name = "Service" };
             serviceControl.GetAuditMessages(Arg.Is(endpoint), Arg.Any<string>(), Arg.Any<int>(), Arg.Any<string>(), Arg.Any<bool>())
-                             .Returns(x => new PagedResult<StoredMessage>
-                             {
-                                 CurrentPage = 1,
-                                 TotalCount = 100,
-                                 Result = new List<StoredMessage>
-                                 {
-                                     new StoredMessage(),
-                                     new StoredMessage()
-                                 }
-                             });
+                .Returns(x => new PagedResult<StoredMessage>
+                {
+                    CurrentPage = 1,
+                    TotalCount = 100,
+                    Result = new List<StoredMessage>
+                    {
+                        new StoredMessage(),
+                        new StoredMessage()
+                    }
+                });
 
             var messageList = messageListFunc();
 
             messageList.Handle(new SelectedExplorerItemChanged(new AuditEndpointExplorerItem(endpoint)));
 
-            eventAggregator.Received(1).PublishOnUIThread(Arg.Any<WorkStarted>());
-            eventAggregator.Received(1).PublishOnUIThread(Arg.Any<WorkFinished>());
```

Comment:
@adamralph That's exactly why I removed this, it's assuming that the internal implementation uses WorkNotifier, which in turn uses EventAggregator. This isn't black box testing and would make the test brittle to changes in WorkNotifer and EventAggregator. Given the test is for making sure the view model calls the servicecontrol instance properly I didn't think also testing it calls WorkNotifier/EventAggregator as well was a good idea.